base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -286,8 +286,9 @@ where B: BlockchainBackend
             // lets load the smt into memory
             let mut smt = blockchain_db.smt_write_access()?;
             warn!(target: LOG_TARGET, "Loading SMT into memory from stored db");
+            let timer = Instant::now();
             *smt = blockchain_db.db_write_access()?.calculate_tip_smt()?;
-            warn!(target: LOG_TARGET, "Finished loading SMT into memory from stored db");
+            warn!(target: LOG_TARGET, "Finished loading SMT into memory from stored db in {:.2?}", timer.elapsed());
```

Comment:
haha yeah, we should go back to caching it.
Doing it every block becomes too intensive, but we should have a checkpoint like every 100 blocks or so,

Reply:
lol yeah just curious if the loading of the SMT was the reason for the base node taking forever to start up. Improvement definitely needed here.

Reply:
Long startup is probably waiting for 5 pings from nodes to decide on a sync peer

Reply:
no it's before that it takes 20s on our test esme network (from the log above)

Reply:
Is it the smt that takes that long?